File: Java(eclipse)/workspace_JavaFX/RockPaperScissors/src/main/java/com/example/RockPaperScissors/GameLogic.java
package com.example.RockPaperScissors;

import java.util.Random;

public class GameLogic {

    private int playerHP;
    private int cpuHP;

    private final GameUI ui;
    private final Random rand;

    public GameLogic(GameUI ui) {
    	
        this.ui = ui;
        rand = new Random();
        reset();
    }

    //////
    //ゲームスタート
    //////
    public void play(String playerHand) {

        String[] hands = {"グー", "チョキ", "パー"};
        String cpuHand = hands[rand.nextInt(3)];

        String handBoth = "[ あなた: " + playerHand + "　CPU: " + cpuHand + " ]  ";

        int result = judge(playerHand, cpuHand);

        if (result == 1) {
        	
            cpuHP--;
            ui.updateCpuHP(cpuHP);
            ui.showMessage(handBoth + "あなたの勝ち！ CPUのHPが減った");
        } else if (result == -1) {
        	
            playerHP--;
            ui.updatePlayerHP(playerHP);
            ui.showMessage(handBoth + "あなたの負け… HPが減った");
            ui.playerDamageEffect();
        } else {
        	
            ui.showMessage(handBoth + "あいこ！");
        }
        //ゲーム継続かの判定
        checkGameEnd(handBoth);
    }

    private int judge(String p, String c) {
    	
        if (p.equals(c)) {
        	
        	return 0;
        }
        if ((p.equals("グー") && c.equals("チョキ")) ||
            (p.equals("チョキ") && c.equals("パー")) ||
            (p.equals("パー") && c.equals("グー"))) {
        	
            return 1;
        }
        return -1;
    }

    private void checkGameEnd(String handBoth) {
    	
        if (playerHP <= 0) {
        	
            ui.showMessage(handBoth + "あなたの負け…");
            ui.showRetryButton();
        }
        if (cpuHP <= 0) {
        	
            ui.showMessage(handBoth + "あなたの勝ち！");
            ui.showRetryButton();
        }
    }
    
    public void reset() {
    	
        playerHP = 3;
        cpuHP = 3;
    }
}

File: 1_C#(VisualStudio)/2_workspace_WPF/1_じゃんけんゲーム_移植元Java/Model/GameLogic.cs
using じゃんけんゲーム_移植元Java.View;

namespace じゃんけんゲーム_移植元Java.Model {
    class GameLogic {

        private int playerHP;
        private int cpuHP;

        private readonly GameUI ui;
        private readonly Random rand;

        public GameLogic(GameUI ui) {

            this.ui = ui;
            rand = new Random();
            Reset();
        }

        // --- ゲームスタート ---
        public void Play(string playerHand) {

            string[] hands = { "グー", "チョキ", "パー" };
            string cpuHand = hands[rand.Next(3)];

            string handBoth = $"[ あなた: {playerHand}　CPU: {cpuHand} ]  ";

            int result = Judge(playerHand, cpuHand);

            if (result == 1) {

                cpuHP--;
                ui.UpdateCpuHP(cpuHP);
                ui.ShowMessage(handBoth + "あなたの勝ち！ CPUのHPが減った");
            } else if (result == -1) {

                playerHP--;
                ui.UpdatePlayerHP(playerHP);
                ui.ShowMessage(handBoth + "あなたの負け… HPが減った");
                ui.PlayerDamageEffect();
            } else {

                ui.ShowMessage(handBoth + "あいこ！");
            }

            CheckGameEnd(handBoth);
        }

        private int Judge(string p, string c) {

            if (p == c)
                return 0;

            if ((p == "グー" && c == "チョキ") ||
                (p == "チョキ" && c == "パー") ||
                (p == "パー" && c == "グー")) {
                return 1;
            }
            return -1;
        }

        private void CheckGameEnd(string handBoth) {

            if (playerHP <= 0) {
                ui.ShowMessage(handBoth + "あなたの負け…");
                ui.ShowRetryButton();
            }

            if (cpuHP <= 0) {
                ui.ShowMessage(handBoth + "あなたの勝ち！");
                ui.ShowRetryButton();
            }
        }

        public void Reset() {

            playerHP = 3;
            cpuHP = 3;
        }
    }
}